# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpChannel.cpp
# Commit: 02933fde7e3f
# Full Hash: 02933fde7e3f3cf4827730cb670dc76b91022813
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2020-10-13 21:43:51
# Regressor Bug: 1665062
# File Overlap Count: 1
# Description:
#   Bug 1665062: HTTPS-Only: Upgraded website creating HTTP auth prompt gets interrupted by error-page r=necko-reviewers,dragana,JulianWels
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D91908
# ==============================================================================

diff -r bc531943d78a -r 02933fde7e3f netwerk/protocol/http/nsHttpChannel.cpp
--- a/netwerk/protocol/http/nsHttpChannel.cpp	Tue Oct 13 18:07:04 2020 +0300
+++ b/netwerk/protocol/http/nsHttpChannel.cpp	Tue Oct 13 14:06:28 2020 +0000
@@ -356,6 +356,7 @@
       mCacheOpenWithPriority(false),
       mCacheQueueSizeWhenOpen(0),
       mCachedContentIsValid(false),
+      mIsAuthChannel(false),
       mAuthRetryPending(false),
       mCachedContentIsPartial(false),
       mCacheOnlyMetadata(false),
@@ -2874,6 +2875,7 @@
       if (rv == NS_ERROR_IN_PROGRESS) {
         // authentication prompt has been invoked and result
         // is expected asynchronously
+        mIsAuthChannel = true;
         mAuthRetryPending = true;
         if (httpStatus == 407 ||
             (mTransaction && mTransaction->ProxyConnectFailed()))
@@ -2903,6 +2905,7 @@
         }
         rv = ProcessNormal();
       } else {
+        mIsAuthChannel = true;
         mAuthRetryPending = true;  // see DoAuthRetry
       }
       break;
@@ -6221,6 +6224,7 @@
   // setting mAuthRetryPending flag and resuming the transaction
   // triggers process of throwing away the unauthenticated data already
   // coming from the network
+  mIsAuthChannel = true;
   mAuthRetryPending = true;
   mProxyAuthPending = false;
   LOG(("Resuming the transaction, we got credentials from user"));
@@ -7249,6 +7253,12 @@
 }
 
 NS_IMETHODIMP
+nsHttpChannel::GetIsAuthChannel(bool* aIsAuthChannel) {
+  *aIsAuthChannel = mIsAuthChannel;
+  return NS_OK;
+}
+
+NS_IMETHODIMP
 nsHttpChannel::SetChannelIsForDownload(bool aChannelIsForDownload) {
   if (aChannelIsForDownload) {
     AddClassFlags(nsIClassOfService::Throttleable);